src/main/java/org/truffleruby/core/array/ArrayNodes.java

```diff
@@ -1258,14 +1257,15 @@ protected ArrayWriteNormalizedNode createWriteNode() {
 
         @Specialization(guards = {
                 "isRubyString(format)",
-                "ropesEqual(format, cachedFormat)"
+                "equalNode.execute(rope(format), cachedFormat)"
```

Comment:
Minor, but I would define `rope()` on StringCachingGuards so StringOperations does not need to be imported.